src/distributions/uniform.rs

```diff
@@ -706,6 +706,75 @@ uniform_simd_int_impl! {
     u8
 }
 
+impl SampleUniform for char {
+    type Sampler = UniformChar;
+}
+
+/// The back-end implementing [`UniformSampler`] for `char`.
+///
+/// Unless you are implementing [`UniformSampler`] for your own type, this type
+/// should not be used directly, use [`Uniform`] instead.
+#[derive(Clone, Copy, Debug)]
+#[cfg_attr(feature = "serde1", derive(Serialize, Deserialize))]
+pub struct UniformChar {
+    sampler: UniformInt<u32>,
+}
+
+impl UniformChar {
+    #[inline]
+    fn new_(mut low: u32, mut high: u32) -> Self {
+        if low >= 0xD800 {
+            low -= 0xE000 - 0xD800;
+        }
```

Comment:
I think it's worth to add the following helper functions:
```rust
/// UTF-16 surrogate range start
const RANGE_START: u32 = 0xD800
/// UTF-16 surrogate range size
const RANGE_SIZE: u32 = 0xE000 - RANGE_START;
/// Convert `char` to compressed `u32`
fn to_comp_u32(c: char) -> u32 {
    match c as u32 {
        c if c >= RANGE_START => c - RANGE_SIZE,
        c => c,
    }
}

/// Convert compressed `u32` to `char`
fn from_comp_u32(mut c: u32) -> u32 {
    if c >= RANE_START {
        c += RANGE_SIZE;
    }
    unsafe { core::char::from_u32_unchecked(c) }
}
```
In addition to make code easier to understand it would allow to remove the `new_` method.